app/models/delete_queue_job.rb

```diff
@@ -171,6 +174,13 @@ def delete_parsed_data(object_id, study_id, *models)
     end
   end
 
+  # ensure there are no orphaned DataArray entries that will cause downstream index violations
+  def delete_orphaned_arrays(study_id)
+    study_file_ids = StudyFile.where(study_id:).pluck(:id)
+    cursor = DataArray.where(study_id:, :study_file_id.nin => study_file_ids)
```

Comment:
I somehow hadn't been aware of [`nin`](https://www.mongodb.com/docs/manual/reference/operator/query/nin/#mongodb-query-op.-nin), i.e. "not in".

Reply:
All of the MongoDB [query operators](https://www.mongodb.com/docs/manual/reference/operator/query-comparison/) are available using the above syntax of `Model.where(:field.operator => query_values`).